File: src/main/java/com/github/stars/service/AiAnalyzeService.java
package com.github.stars.service;

import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.github.stars.entity.GithubRepo;
import com.github.stars.mapper.GithubRepoMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.http.*;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import javax.annotation.Resource;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicInteger;

/**
 * AI 分析总结服务
 * 将筛选出的项目集合发送给 DeepSeek 进行智能分析和汇总
 */
@Service
public class AiAnalyzeService {

    private static final Logger log = LoggerFactory.getLogger(AiAnalyzeService.class);
    private static final int MAX_REPOS = 30; // 测试阶段限制30个

    @Resource
    private GithubRepoMapper githubRepoMapper;

    @Autowired
    @Qualifier("longTimeoutRestTemplate")
    private RestTemplate restTemplate;

    @Resource
    private ObjectMapper objectMapper;

    @Resource
    private SystemConfigService configService;

    /** 分析任务结果缓存 */
    private final Map<String, String> results = new ConcurrentHashMap<>();
    private final Map<String, String> statuses = new ConcurrentHashMap<>();
    private final AtomicInteger taskCounter = new AtomicInteger(0);

    /**
     * 创建分析任务并异步执行
     * @return 任务ID
     */
    public String createAnalyzeTask(String keyword, String language, String categoryIds,
                                     String sortBy, String sortOrder) {
        String taskId = "analyze_" + taskCounter.incrementAndGet();
        statuses.put(taskId, "PROCESSING");

        // 异步执行（用 CompletableFuture 确保不阻塞）
        final String tid = taskId;
        CompletableFuture.runAsync(() -> executeAnalyze(tid, keyword, language, categoryIds, sortBy, sortOrder));

        return taskId;
    }

    /**
     * 创建趋势分析任务（分析 GitHub trending repos 的技术趋势）
     */
    public String createTrendingAnalyzeTask(String since, String language,
                                             List<Map<String, Object>> repos) {
        String taskId = "trending_" + taskCounter.incrementAndGet();
        statuses.put(taskId, "PROCESSING");
        final String tid = taskId;
        CompletableFuture.runAsync(() -> executeTrendingAnalyze(tid, since, language, repos));
        return taskId;
    }

    private void executeTrendingAnalyze(String taskId, String since, String language,
                                         List<Map<String, Object>> repos) {
        try {
            if (repos.isEmpty()) {
                results.put(taskId, "### 暂无数据\n\n当前没有趋势数据可供分析。");
                statuses.put(taskId, "COMPLETED");
                return;
            }

            StringBuilder sb = new StringBuilder();
            sb.append("你是一位顶尖的技术趋势分析师。以下是 GitHub 上 ");
            sb.append(since.equals("daily") ? "今日" : since.equals("weekly") ? "本周" : "本月");
            if (language != null && !language.isEmpty()) sb.append(" ").append(language).append(" 语言");
            sb.append("最热门的新项目（共 ").append(repos.size()).append(" 个）。\n\n");

            for (int i = 0; i < repos.size(); i++) {
                Map<String, Object> r = repos.get(i);
                sb.append(i + 1).append(". ").append(r.get("full_name")).append("\n");
                String desc = (String) r.get("description");
                if (desc != null && !desc.isEmpty()) {
                    sb.append("   描述: ").append(desc.length() > 150 ? desc.substring(0, 150) + "..." : desc).append("\n");
                }
                sb.append("   ⭐").append(r.get("stargazers_count")).append(" | 语言: ").append(r.get("language")).append("\n\n");
            }

            sb.append("请用中文输出一份趋势分析报告（Markdown格式），包含:\n");
            sb.append("### 一、热门方向\n列出 3-5 个当前最热的技术方向，每个方向一句话概括\n\n");
            sb.append("### 二、项目用途分类\n按用途分类这些新项目（如AI工具、开发框架、DevOps工具、安全、数据等）\n\n");
            sb.append("### 三、值得关注的项目\n推荐 3 个最值得关注的仓库，说明理由\n\n");
            sb.append("### 四、趋势洞察\n分析这些新项目反映的技术趋势和开发者关注点的变化\n\n");
            sb.append("只输出分析报告，不要开头语结尾语。");

            log.info("趋势分析 prompt: {} chars, {} repos", sb.length(), repos.size());
            String analysis = callDeepSeek(sb.toString());
            if (analysis == null) analysis = "### 分析失败\n\nAI 服务暂时不可用，请稍后重试。";
            results.put(taskId, analysis);
            statuses.put(taskId, "COMPLETED");
        } catch (Exception e) {
            log.error("趋势分析失败", e);
            results.put(taskId, "### 分析失败\n\n" + e.getMessage());
            statuses.put(taskId, "COMPLETED");
        }
    }

    /**
     * 获取任务状态
     */
    public Map<String, Object> getTaskStatus(String taskId) {
        Map<String, Object> result = new LinkedHashMap<>();
        String status = statuses.getOrDefault(taskId, "NOT_FOUND");
        result.put("taskId", taskId);
        result.put("status", status);
        if ("COMPLETED".equals(status)) {
            result.put("content", results.get(taskId));
        }
        return result;
    }

    private void executeAnalyze(String taskId, String keyword, String language,
                                String categoryIds, String sortBy, String sortOrder) {
        try {
            // 1. 查询符合条件的项目（最多30个）
            List<GithubRepo> repos = queryRepos(keyword, language, categoryIds, sortBy, sortOrder);
            if (repos.isEmpty()) {
                statuses.put(taskId, "COMPLETED");
                results.put(taskId, "### 分析结果\n\n当前筛选条件下没有找到任何项目。");
                return;
            }

            log.info("AI分析任务 {}: 收集到 {} 个项目", taskId, repos.size());

            // 2. 构建 prompt
            String prompt = buildAnalyzePrompt(repos);

            // 3. 调用 DeepSeek
            log.info("AI分析 prompt 大小: {} 字符", prompt.length());
            String analysis;
            try {
                analysis = callDeepSeek(prompt);
            } catch (Exception e) {
                log.error("DeepSeek 调用异常", e);
                String msg = e.getMessage() != null ? e.getMessage() : "未知错误";
                if (msg.contains("timed out") || msg.contains("timeout")) {
                    analysis = "### 分析超时\n\nAI 服务响应超时（当前筛选项目数据量较大）。建议：\n- 缩小筛选范围（使用关键词或分类筛选）\n- 减少项目数量后重试";
                } else {
                    analysis = "### 分析失败\n\nAI 服务调用异常: " + msg;
                }
                results.put(taskId, analysis);
                statuses.put(taskId, "COMPLETED");
                return;
            }
            if (analysis == null || analysis.isEmpty()) {
                analysis = "### 分析失败\n\nAI 服务返回空结果，请稍后重试。";
            }

            // 4. 保存结果
            results.put(taskId, analysis);
            statuses.put(taskId, "COMPLETED");
            log.info("AI分析任务 {} 完成, 结果长度: {}", taskId, analysis.length());

        } catch (Exception e) {
            log.error("AI分析任务 {} 失败", taskId, e);
            statuses.put(taskId, "COMPLETED");
            results.put(taskId, "### 分析失败\n\n" + e.getMessage());
        }
    }

    /**
     * 按筛选条件查询项目（最多 MAX_REPOS 个）
     */
    private List<GithubRepo> queryRepos(String keyword, String language, String categoryIds,
                                         String sortBy, String sortOrder) {
        LambdaQueryWrapper<GithubRepo> wrapper = new LambdaQueryWrapper<>();

        // 分类筛选
        if (categoryIds != null && !categoryIds.isEmpty()) {
            wrapper.inSql(GithubRepo::getId,
                "SELECT repo_id FROM repo_category WHERE category_id IN (" + categoryIds + ")");
        }

        // 关键词搜索
        if (keyword != null && !keyword.isEmpty()) {
            wrapper.and(w -> w
                .like(GithubRepo::getRepoName, keyword)
                .or().like(GithubRepo::getDescription, keyword)
                .or().like(GithubRepo::getOwnerName, keyword)
                .or().like(GithubRepo::getFullName, keyword));
        }

        // 多语言筛选
        if (language != null && !language.isEmpty()) {
            List<String> langs = Arrays.asList(language.split(","));
            if (!langs.contains("")) wrapper.in(GithubRepo::getLanguage, langs);
        }

        // 排序
        boolean isAsc = "asc".equalsIgnoreCase(sortOrder);
        if ("stars_count".equals(sortBy)) {
            wrapper.orderBy(true, isAsc, GithubRepo::getStarsCount);
        } else if ("forks_count".equals(sortBy)) {
            wrapper.orderBy(true, isAsc, GithubRepo::getForksCount);
        } else {
            wrapper.orderBy(true, isAsc, GithubRepo::getStarredAt);
        }

        wrapper.last("LIMIT " + MAX_REPOS);
        return githubRepoMapper.selectList(wrapper);
    }

    /**
     * 构建 AI 分析 prompt
     */
    private String buildAnalyzePrompt(List<GithubRepo> repos) {
        StringBuilder sb = new StringBuilder();
        sb.append("你是一位资深的 GitHub 开源项目分析专家。请对以下 ").append(repos.size())
          .append(" 个 GitHub 项目进行全面分析和总结。\n\n");

        // 项目列表
        sb.append("## 项目列表\n\n");
        for (int i = 0; i < repos.size(); i++) {
            GithubRepo r = repos.get(i);
            sb.append("### ").append(i + 1).append(". ").append(r.getFullName()).append("\n");
            sb.append("- **语言**: ").append(r.getLanguage() != null ? r.getLanguage() : "未知").append("\n");
            sb.append("- **Star**: ").append(r.getStarsCount())
              .append(" | Fork: ").append(r.getForksCount()).append("\n");
            if (r.getDescriptionCn() != null && !r.getDescriptionCn().isEmpty()) {
                String descCn = r.getDescriptionCn().length() > 200
                    ? r.getDescriptionCn().substring(0, 200) + "..."
                    : r.getDescriptionCn();
                sb.append("- **描述**: ").append(descCn).append("\n");
            } else if (r.getDescription() != null && !r.getDescription().isEmpty()) {
                String desc = r.getDescription().length() > 200
                    ? r.getDescription().substring(0, 200) + "..."
                    : r.getDescription();
                sb.append("- **描述**: ").append(desc).append("\n");
            }
            // README 摘要（前200字符，精简以加快AI响应）
            if (r.getReadmeCn() != null && !r.getReadmeCn().isEmpty()) {
                String readme = r.getReadmeCn().length() > 200
                    ? r.getReadmeCn().substring(0, 200) + "..."
                    : r.getReadmeCn();
                sb.append("- **README**: ").append(readme.replace("\n", " ")).append("\n");
            }
            sb.append("\n");
        }

        sb.append("## 分析要求\n\n");
        sb.append("请用中文输出一份结构化的分析报告，使用 Markdown 格式。要求包含以下部分：\n\n");
        sb.append("### 一、总体概览\n");
        sb.append("- 项目总数、总 Star 数、总 Fork 数\n");
        sb.append("- 项目整体质量评估（活跃度、社区参与度等）\n\n");
        sb.append("### 二、技术栈分析\n");
        sb.append("- 编程语言分布（按数量排序）\n");
        sb.append("- 主流技术框架和工具\n");
        sb.append("- 技术趋势观察\n\n");
        sb.append("### 三、应用场景分类\n");
        sb.append("- 将项目按应用场景归类（如 AI/机器学习、Web开发、DevOps、安全、工具等）\n");
        sb.append("- 每个类别列出代表性项目\n\n");
        sb.append("### 四、热门项目 TOP 5\n");
        sb.append("- 按 Star 数排名的前 5 个项目，简要说明其特点和受欢迎原因\n\n");
        sb.append("### 五、趋势与洞察\n");
        sb.append("- 近期热门方向（结合项目描述和 README 分析）\n");
        sb.append("- 值得关注的优质项目推荐\n\n");
        sb.append("### 六、总结建议\n");
        sb.append("- 对这些项目的整体评价\n");
        sb.append("- 推荐优先关注的项目（2-3个）\n\n");
        sb.append("注意：只输出分析报告本身，不要任何开头语或结尾语。");

        return sb.toString();
    }

    /**
     * 调用 DeepSeek API
     */
    private String callDeepSeek(String prompt) {
        try {
            ObjectNode requestBody = objectMapper.createObjectNode();
            requestBody.put("model", configService.getValue("deepseek.model", "deepseek-chat"));
            requestBody.put("temperature", 0.3);
            requestBody.put("max_tokens", 32768);

            ArrayNode messages = objectMapper.createArrayNode();
            ObjectNode sysMsg = objectMapper.createObjectNode();
            sysMsg.put("role", "system");
            sysMsg.put("content", "你是一位资深的开源项目分析师，擅长从大量项目中提取关键洞察。请用中文输出专业的分析报告。");
            messages.add(sysMsg);

            ObjectNode userMsg = objectMapper.createObjectNode();
            userMsg.put("role", "user");
            userMsg.put("content", prompt);
            messages.add(userMsg);
            requestBody.set("messages", messages);

            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            headers.set("Authorization", "Bearer " + configService.getValue("deepseek.api_key"));

            HttpEntity<String> entity = new HttpEntity<>(objectMapper.writeValueAsString(requestBody), headers);
            ResponseEntity<String> response = restTemplate.postForEntity(
                configService.getValue("deepseek.api_url", "https://api.deepseek.com/v1/chat/completions"),
                entity, String.class);

            if (response.getStatusCode() == HttpStatus.OK && response.getBody() != null) {
                JsonNode root = objectMapper.readTree(response.getBody());
                JsonNode choices = root.get("choices");
                if (choices != null && choices.isArray() && choices.size() > 0) {
                    String content = choices.get(0).get("message").get("content").asText();
                    log.info("AI 分析返回内容长度: {}", content.length());
                    return content.trim();
                }
                log.error("AI 分析响应格式异常: {}", response.getBody().substring(0, Math.min(500, response.getBody().length())));
            } else {
                log.error("AI 分析 HTTP 状态异常: {} body长度={}", response.getStatusCode(),
                    response.getBody() != null ? response.getBody().length() : 0);
            }
            return null;
        } catch (Exception e) {
            log.error("AI 分析请求失败", e);
            return null;
        }
    }
}

File: backend-nestjs/src/ai/services/ai-analyze.service.ts
import { Injectable, Logger, OnModuleInit } from '@nestjs/common';
import { PrismaService } from '../../prisma/prisma.service';
import { ConfigService } from '../../config/config.service';

const MAX_REPOS = 30;

@Injectable()
export class AiAnalyzeService implements OnModuleInit {
    private readonly logger = new Logger(AiAnalyzeService.name);
    private counter = 0;

    constructor(
        private readonly prisma: PrismaService,
        private readonly config: ConfigService,
    ) {}

    /**
     * 模块初始化 — 从数据库恢复 counter 计数器（避免应用重启后任务 ID 冲突）
     *
     * 查询最新的 analyze_ 前缀任务，从中解析序号并设置 counter。
     */
    async onModuleInit() {
        const latest = await this.prisma.aiAnalyzeTask.findFirst({
            where: { taskId: { startsWith: 'analyze_' } },
            orderBy: { createdAt: 'desc' },
            select: { taskId: true },
        });
        if (latest) {
            const match = latest.taskId.match(/analyze_(\d+)/);
            if (match) this.counter = parseInt(match[1], 10);
        }
    }

    /**
     * 根据筛选条件查询仓库列表（最多 {MAX_REPOS} 条）
     *
     * 支持按分类 ID、关键词、语言筛选，按 star 数 / fork 数 / starredAt 排序。
     *
     * @param params.keyword 关键词（模糊匹配仓库名、描述、所有者、全名）
     * @param params.language 编程语言（逗号分隔）
     * @param params.categoryIds 分类 ID（逗号分隔）
     * @param params.sortBy 排序字段（stars_count / forks_count / 其他）
     * @param params.sortOrder 排序方向（asc / desc）
     * @returns 匹配的仓库列表
     */
    private async queryRepos(params: { keyword?: string; language?: string; categoryIds?: string; sortBy?: string; sortOrder?: string }) {
        const AND: any[] = [];
        if (params.categoryIds) {
            const ids = params.categoryIds
                .split(',')
                .map(Number)
                .filter((n) => !isNaN(n))
                .map(BigInt);
            AND.push({ repoCategories: { some: { categoryId: { in: ids } } } });
        }
        if (params.keyword) {
            const kw = params.keyword;
            AND.push({
                OR: [
                    { repoName: { contains: kw } },
                    { description: { contains: kw } },
                    { ownerName: { contains: kw } },
                    { fullName: { contains: kw } },
                ],
            });
        }
        if (params.language) {
            const langs = params.language.split(',').filter(Boolean);
            if (langs.length) AND.push({ language: { in: langs } });
        }
        const where: any = AND.length ? { AND } : {};
        const sortField = params.sortBy === 'stars_count' ? 'starsCount' : params.sortBy === 'forks_count' ? 'forksCount' : 'starredAt';
        const sortDir = params.sortOrder === 'asc' ? 'asc' : 'desc';
        return this.prisma.githubRepo.findMany({ where, orderBy: { [sortField]: sortDir }, take: MAX_REPOS });
    }

    /**
     * 构建 AI 分析提示词
     *
     * 将仓库列表拼接为结构化文本，包含序号、名称、语言、star/fork 数、描述和 README 摘要。
     *
     * @param repos 仓库列表
     * @returns 分析提示词字符串
     */
    private buildAnalyzePrompt(repos: any[]) {
        let list = '';
        repos.forEach((r, i) => {
            const desc = String(r.descriptionCn || r.description || '').substring(0, 200);
            const readme = String(r.readmeCn || r.readmeOriginal || '').substring(0, 200);
            list += `${i + 1}. **${r.repoName || r.fullName}** (${r.language || '未知'}, ⭐${r.starsCount}, Fork:${r.forksCount})\n`;
            list += `   描述: ${desc}\n`;
            if (readme) list += `   README: ${readme}\n`;
            list += '\n';
        });
        return `请分析以下 GitHub 项目集合：

${list}

请输出结构化报告：1.总体概览 2.技术栈分析 3.应用场景分类 4.热门项目TOP5 5.趋势与洞察 6.总结建议

【重要】用中文输出，直接开始正文，不要加开头语（如"好的"）或结尾语（如"以上是分析"）。`;
    }

    /**
     * 调用 DeepSeek API 执行 AI 分析
     *
     * 从系统配置中读取 API Key、URL、模型名，发送 chat completions 请求。
     * 返回 AI 回复的文本内容，若失败则返回错误描述字符串。
     *
     * @param prompt 用户提示词
     * @returns AI 生成的文本内容，或错误描述（不为 null）
     */
    private async callDeepSeek(prompt: string): Promise<string | null> {
        const apiKey = await this.config.getValue('deepseek.api_key');
        const apiUrl = await this.config.getValueDefault('deepseek.api_url', 'https://api.deepseek.com/v1/chat/completions');
        const model = await this.config.getValueDefault('deepseek.model', 'deepseek-chat');
        if (!apiKey) return 'DeepSeek API Key 未配置';
        try {
            const res = await fetch(apiUrl, {
                method: 'POST',
                headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${apiKey}` },
                body: JSON.stringify({
                    model,
                    temperature: 0.3,
                    max_tokens: 32768,
                    messages: [
                        { role: 'system', content: '你是专业的代码分析师。' },
                        { role: 'user', content: prompt },
                    ],
                }),
            });
            if (!res.ok) return `AI 服务异常 (${res.status})`;
            const data = await res.json();
            return data.choices?.[0]?.message?.content?.trim() || null;
        } catch (e) {
            const msg = e instanceof Error ? e.message : String(e);
            this.logger.error('DeepSeek AI 分析调用失败: ' + msg);
            return msg.includes('timeout') ? '分析超时' : 'AI 异常: ' + msg;
        }
    }

    /**
     * 执行分析任务的内部方法（异步，不阻塞 API 响应）
     *
     * 流程：查询仓库 → 构建提示词 → 调用 DeepSeek → 更新任务结果。
     * 无论成功或失败，都会将任务状态更新为 COMPLETED。
     *
     * @param taskId 任务 ID
     * @param keyword 关键词筛选
     * @param language 语言筛选
     * @param categoryIds 分类筛选
     * @param sortBy 排序字段
     * @param sortOrder 排序方向
     */
    private async executeAnalyze(
        taskId: string,
        keyword: string,
        language: string,
        categoryIds: string,
        sortBy: string,
        sortOrder: string,
    ) {
        try {
            this.logger.log('分析任务开始: taskId=' + taskId);
            const repos = await this.queryRepos({ keyword, language, categoryIds, sortBy, sortOrder });
            if (!repos.length) {
                this.logger.log('分析任务完成（无数据）: taskId=' + taskId);
                await this.prisma.aiAnalyzeTask.update({
                    where: { taskId },
                    data: { status: 'COMPLETED', content: '没有找到任何项目', finishedAt: new Date() },
                });
                return;
            }
            const prompt = this.buildAnalyzePrompt(repos);
            const result = await this.callDeepSeek(prompt);
            await this.prisma.aiAnalyzeTask.update({
                where: { taskId },
                data: { status: 'COMPLETED', content: result || 'AI 返回空结果', finishedAt: new Date() },
            });
            this.logger.log('分析任务完成: taskId=' + taskId);
        } catch (e) {
            this.logger.error('分析任务执行异常: taskId=' + taskId + ', 错误=' + (e instanceof Error ? e.message : String(e)));
            await this.prisma.aiAnalyzeTask.update({
                where: { taskId },
                data: { status: 'COMPLETED', content: '分析失败: ' + (e instanceof Error ? e.message : String(e)), finishedAt: new Date() },
            });
        }
    }

    /**
     * 创建 AI 分析任务
     *
     * 创建一条 PROCESSING 状态的分析任务记录，然后异步执行分析（不阻塞 HTTP 响应）。
     *
     * @param keyword 关键词筛选
     * @param language 语言筛选
     * @param categoryIds 分类 ID 筛选
     * @param sortBy 排序字段
     * @param sortOrder 排序方向
     * @returns 任务 ID
     */
    async createAnalyzeTask(keyword: string, language: string, categoryIds: string, sortBy: string, sortOrder: string) {
        const taskId = 'analyze_' + ++this.counter;
        this.logger.log('创建分析任务: taskId=' + taskId);
        await this.prisma.aiAnalyzeTask.create({
            data: {
                taskId,
                type: 'analyze',
                status: 'PROCESSING',
                params: JSON.stringify({ keyword, language, categoryIds, sortBy, sortOrder }),
                createdAt: new Date(),
            },
        });
        this.executeAnalyze(taskId, keyword, language, categoryIds, sortBy, sortOrder).catch((e) =>
            this.logger.error('分析任务执行失败', e),
        );
        return taskId;
    }

    /**
     * 创建 Trending 趋势分析任务
     *
     * 对 GitHub Trending 爬取结果进行 AI 分析，生成趋势报告。
     * 若无数据则直接标记为 COMPLETED；否则异步执行分析。
     *
     * @param since 时间范围（daily / weekly / monthly）
     * @param language 语言筛选
     * @param repos Trending 仓库列表
     * @returns 任务 ID
     */
    async createTrendingAnalyzeTask(since: string, language: string, repos: any[]) {
        const taskId = 'trending_' + ++this.counter;
        const period = { daily: '今日', weekly: '本周', monthly: '本月' }[since] || since;
        this.logger.log('创建趋势分析任务: taskId=' + taskId + ', since=' + since + ', reposCount=' + repos.length);

        if (!repos.length) {
            await this.prisma.aiAnalyzeTask.create({
                data: { taskId, type: 'trending', status: 'COMPLETED', content: '暂无数据', createdAt: new Date(), finishedAt: new Date() },
            });
            return taskId;
        }

        await this.prisma.aiAnalyzeTask.create({
            data: {
                taskId,
                type: 'trending',
                status: 'PROCESSING',
                params: JSON.stringify({ since, language }),
                createdAt: new Date(),
            },
        });

        const list = repos
            .map((r: any, i: number) => `${i + 1}. **${r.fullName}** (⭐${r.starsCount}, ${r.language || '未知'})`)
            .join('\n');
        const prompt = `分析 GitHub ${period}趋势：\n\n${language ? '语言: ' + language + '\n' : ''}${list}\n\n分析热门方向、用途分类、最值得关注的3个项目、趋势洞察。\n\n【重要】直接开始正文，不要加开头语或结尾语。`;

        (async () => {
            try {
                const r = await this.callDeepSeek(prompt);
                await this.prisma.aiAnalyzeTask.update({
                    where: { taskId },
                    data: { status: 'COMPLETED', content: r || '分析失败', finishedAt: new Date() },
                });
                this.logger.log('趋势分析任务完成: taskId=' + taskId);
            } catch (e) {
                this.logger.error('趋势分析任务异常: taskId=' + taskId + ', 错误=' + (e instanceof Error ? e.message : String(e)));
                await this.prisma.aiAnalyzeTask.update({
                    where: { taskId },
                    data: { status: 'COMPLETED', content: '分析失败', finishedAt: new Date() },
                });
            }
        })().catch((e) => this.logger.error('趋势分析任务失败', e));

        return taskId;
    }

    /**
     * 查询任务状态（P0 FIX: 从数据库读取，非内存）
     *
     * @param taskId 任务 ID
     * @returns 任务状态信息，包含 status、content（完成时）等
     */
    async getTaskStatus(taskId: string) {
        const task = await this.prisma.aiAnalyzeTask.findUnique({ where: { taskId } });
        if (!task) return { success: false, taskId, status: 'NOT_FOUND' };
        return {
            success: true,
            taskId: task.taskId,
            status: task.status,
            content: task.status === 'COMPLETED' ? task.content : undefined,
        };
    }
}
